### PR TITLE
NMS: Fix buffer size calculation bug

### PR DESCRIPTION
**Pull request by andrejsokolov**
Correct method of temporary buffer size calculation
ticket -43347

### CODE REVIEW

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp [hunk 1]
```diff
@@ -102,7 +95,11 @@ bool isCMXEnough(int cmxSize, int numSlices, std::vector<int> bufferSizes) {
         return true;
     };
 
-    return std::all_of(bufferSizes.begin(), bufferSizes.end(), buffer_allocate);
+    return (numSlices > 0) && std::all_of(bufferSizes.begin(), bufferSizes.end(), buffer_allocate);
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:98]: What's happening when numSlices == 0?

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp [hunk 2]
```diff
@@ -102,7 +95,11 @@ bool isCMXEnough(int cmxSize, int numSlices, std::vector<int> bufferSizes) {
...
+}
+
+bool isOneSliceEnough(int cmxSize, std::vector<int> bufferSizes) {
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:101]: Nitpick: take vector by const ref

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp (hunk 1 above)
Comment (andrejsokolov) [reply inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:98]: the function can return `true`, but should return `false`

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:98]: Thanks

**Review by AndrewBakalinIntel — APPROVED** [review]

**Review by gladilov-gleb — APPROVED** [review]

**Pull request merged**